### PR TITLE
doc : Add documentation for using jkube snapshots

### LINKED ISSUE #2254: Document the usage of SNAPSHOTs
### Description

After #2253, SNAPSHOTs are now being published nightly:
- https://oss.sonatype.org/content/repositories/snapshots/org/eclipse/jkube/
- https://ci.eclipse.org/jkube/job/ReleasePipelines/job/ReleaseSnapshots/

We need to add documentation for users to learn how to use these SNAPSHOTs. Maybe also tune the Quickstarts or provide a specific quickstart for this purpose (replacement of #2069)

### PR DESCRIPTION
**Pull request by Prafful-Rastogi**
## Description

Fixes #2254

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] My code follows the style guidelines of this project
 - [ ] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: USING-SNAPSHOT-ARTIFACTS.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+- +Our [Jenkins Snapshot release pipeline](https://ci.eclipse.org/jkube/job/ReleasePipelines/job/ReleaseSnapshots/) updates SNAPSHOT artifacts every night.
+-
+- +## Using SNAPSHOTs in Maven Project
+  +In order to use these artifacts, update your `pom.xml` with these:
+- +```xml +<pluginRepositories>
+- <pluginRepository>
+- <id>oss.sonatype.org</id>
+- <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+- <snapshots>
+-      <enabled>true</enabled>
```
Comment (rohanKanojia) [inline USING-SNAPSHOT-ARTIFACTS.md:13]: ah, sorry for not being clear. You're supposed to not add preceding `-`/`+` signs. They were added by `git diff` output.

Comment (rohanKanojia) [reply USING-SNAPSHOT-ARTIFACTS.md:13]: You can view preview of your work here https://github.com/Prafful-Rastogi/jkube/blob/pr/issue2254/USING-SNAPSHOT-ARTIFACTS.md

Comment (Prafful-Rastogi) [reply USING-SNAPSHOT-ARTIFACTS.md:13]: Hi @rohanKanojia thats fine will make changes and let you know.

File: README.md [hunk 2]
```diff
@@ -29,10 +30,10 @@
     - [Troubleshooting](#troubleshooting)
 - [Rebranding Notice](#rebranding-notice--loudspeaker-)
 - [Contributing](https://www.eclipse.org/jkube/contributing/)
+- [How to use Eclipse JKube snapshot artifacts?](./USING-JKUBE-SNAPSHOTS.md)
```
Comment (rohanKanojia) [inline README.md:33]:
```suggestion
- [How to use Eclipse JKube snapshot artifacts?](./USING-SNAPSHOT-ARTIFACTS.md)
```

Comment (rohanKanojia) [reply README.md:33]: Sorry, I did a mistake while adding steps. There was a typo in entry (file name was wrong)

Comment (Prafful-Rastogi) [reply README.md:33]: No worries will update it and push it again.

Comment (Prafful-Rastogi) [reply README.md]: Updated from my end. Please check it and let me know if there is anything to be done.

File: README.md [hunk 3]
```diff
@@ -29,10 +30,10 @@
     - [Troubleshooting](#troubleshooting)
 - [Rebranding Notice](#rebranding-notice--loudspeaker-)
 - [Contributing](https://www.eclipse.org/jkube/contributing/)
+- [How to use Eclipse JKube snapshot artifacts?](./USING-SNAPSHOT-SNAPSHOTS.md)
```
Comment (rohanKanojia) [inline README.md:33]: Looks like this hasn't been updated correctly.

```suggestion
- [How to use Eclipse JKube snapshot artifacts?](./USING-SNAPSHOT-ARTIFACTS.md)
```

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**